src/System.Management.Automation/engine/EventManager.cs

```diff
@@ -2042,6 +2042,19 @@ private ScriptBlock CreateBoundScriptBlock(ScriptBlock scriptAction)
 
         #region IComparable<PSEventSubscriber> Members
 
+        /// <summary>
+        /// Determines whether the specified object is equal to the current object.
+        /// </summary>
+        /// <param name="obj">The object to compare with the current object.</param>
+        /// <returns>
+        /// <see langword="true"/> if the specified object is equal to the current object;
+        /// otherwise, <see langword="false"/>.
+        /// </returns>
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as PSEventSubscriber);
```

Comment:
From the doc you linked the suggested form was something like this instead. Is there a significant difference either way? 🤔 

```suggestion
            return obj is PSEventSubscriber es && Equals(es);
```

Reply:
Currently the method replies on the null check in the `Equals(PSEventSubscriber)` method immediately below for correctness, I guess your suggested change makes sense though.

https://github.com/PowerShell/PowerShell/blob/72b8c61c7d74c9631b80a05223d4bc01852ab884/src/System.Management.Automation/engine/EventManager.cs#L2064-L2072